registry/storage/driver/s3-goamz/s3.go

```diff
@@ -573,6 +573,17 @@ func getPermissions() s3.ACL {
 	return s3.Private
 }
 
+// mustV4Auth checks whether must use v4 auth in specific region.
+// Please see documentation at http://docs.aws.amazon.com/general/latest/gr/signature-version-2.html
```

Comment:
That is great! ping @aaronlehmann @dmcgowan PTAL.
Wish that we could make this move on.